functions/imgs_to_gif.m

```diff
@@ -2,10 +2,19 @@ function images_to_gif(imgs)
   date_str = datestr(now,'yyyymmddTHHMMSS');
   filename = [date_str '.gif'];
   for t=1:size(imgs,3)
-      if t==1;
-        imwrite(imgs(:,:,t)./12,filename,'gif', 'DelayTime',0.5, 'Loopcount',inf);
-      else
-        imwrite(imgs(:,:,t)./12,filename,'gif', 'DelayTime',0.5, 'WriteMode','append');
-      end
+      [imind,cm] = rgb2ind(squeeze(imgs(:,:,t,:)),256);
+        if t == 1;
+          imwrite(imind,cm,filename,'gif', 'DelayTime',0.4, 'Loopcount',inf);
+        else
+           imwrite(imind,cm,filename,'gif', 'DelayTime',0.4, 'WriteMode','append');
+        end
+      
+      
+%       if t==1;
+%         imwrite(imgs(:,:,t,:)./12,filename,'gif', 'DelayTime',0.5, 'Loopcount',inf);
+%       else
+%         imwrite(imgs(:,:,t,:)./12,filename,'gif', 'DelayTime',0.5, 'WriteMode','append');
+%       end
```

Comment:
Please create a new file called "colour_imgs_to_gif.m" and move all this code there except the comment. Then remove the comments and "imgs_to_gif.m" will be the version that works on grey images. Thanks!